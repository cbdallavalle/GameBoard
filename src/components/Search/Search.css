.Search form {
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: flex-start;
  align-content: center;
}

.Search input {
  padding-left: 15px;
  min-width: 290px;
  width: 83%;
  max-width: 332px;
  height: 36.4px;
}

.Search button {
  width: 80px;
  height: 42px;
  margin: 0;
  border: 2px solid black;
  font-size: 17px;
  font-family: 'Open Sans', sans-serif;
}

.Search h1 {
  font-size: 58px;
}