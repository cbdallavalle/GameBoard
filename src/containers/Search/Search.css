.Search {
  width: 700px;
  min-height: 100vh;
  padding: 25px 0;
  font-family: 'Open Sans', sans-serif;
}

.game-description {
  position: relative;
  height: fit-content;
  width: 600px;
  margin: 50px auto;
  overflow-y: scroll;
  background-color: #f4f4f4a6;
  padding: 8px 16px;
}

.fa-add {
  color: #0da50d;
  margin-right: 10px;
}

.no-results {
  font-size: 27px;  
  font-weight: bold;
  color: #525252;
  margin-top: 150px;
}

.Search form {
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: flex-start;
  align-content: center;
  height: fit-content;
}

.Search button {
  width: 100px;
  height: 42px;
  margin: 0;
  border: 2px solid #313131;
  font-size: 17px;
  margin-top: 14px;
  background-color: #dfbaa7;
  border-left: none;
}

.Search h1 {
  font-size: 58px;
}

.search-results h3 {
  padding: 25px 10px;
  cursor: pointer;
  text-align: left;
  font-family: 'Open Sans', sans-serif;
}

.search-results h3:hover {
  background-color: #8080804a;
}

.search-results {
  width: 600px;
  height: 400px;
  margin: auto;
  overflow-y: scroll;
  background-color: #ffffffcf;
}

#not-added {
  position: absolute;
  top: 11px;
  right: 25px;
  border: none;
  border-radius: 120px;
  width: 43px;
  background-color: #ffffff00;
  background-image: url('../../assets/plus.svg');
  background-repeat: no-repeat;
  background-position: center;
}

#not-added:hover {
  cursor: pointer;
  background-color: white;
  border: none;
}

#added {
  position: absolute;
  top: 11;
  right: 25px;
  border: none;
  border-radius: 120px;
  width: 43px;
  transition: background .3s linear;
  background-image: url('../../assets/success.svg');
  background-position: center;
  background-size: cover;
  background-color: #ffffff00;
}

#loading {
  width: 100px;
}

#loading-cont {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: space-evenly;
  width: 600px;
  height: 400px;
  position: absolute;
  top: 21px;
  left: 17px;
  background-color: #bfbfbf91;
  z-index: 100;
}