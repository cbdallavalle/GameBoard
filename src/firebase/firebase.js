import * as firebase from 'firebase';

const config = {
  apiKey: "AIzaSyBDU0_Pa8xdAWKmtKKfo3QkbrtR56cl024",
  authDomain: "dallavalle-gameboard.firebaseapp.com",
  databaseURL: "https://dallavalle-gameboard.firebaseio.com",
  projectId: "dallavalle-gameboard",
  storageBucket: "",
  messagingSenderId: "250758219123"
};

if(!firebase.apps.length) {
  firebase.initializeApp(config)
}

const auth = firebase.auth();

export {
  auth,
}