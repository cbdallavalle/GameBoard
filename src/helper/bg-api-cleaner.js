export const cleanSearch = (searches) => {
  return searches.elements[0].elements.map( search => {
    const id = search.attributes.id;
    const name = search.elements[0].attributes.value;
    return {id, name}
  })
}

export const cleanGameDetails = game => {
  return game.elements[0].elements.map( gameObj => {
    const gameDetails =  gameObj.elements.reduce( (cleanedGame, gameDetail) => {
      if (gameDetail.name === 'thumbnail' || gameDetail.name === 'image' || gameDetail.name === 'description') {
        cleanedGame[gameDetail.name] = gameDetail.elements[0].text
      }
      return cleanedGame
    }, {} )
    const id = gameObj.attributes.id;
    return {...gameDetails, id}
  })
}

export const fetchBoardGames = async url => {
  const response = await fetch(url);
  const responseText = await response.text();

  const convert = require('xml-js');
  const options = {ignoreComment: true, alwaysChildren: true };
  const result = convert.xml2js(responseText, options);
  return result
}