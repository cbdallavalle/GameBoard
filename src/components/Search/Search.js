import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

import { db } from '../../firebase';
import * as api from '../../helper/bg-api-cleaner';
import Card from '../Card/Card';
import { updateFavorites } from '../../actions';
import loading from '../../assets/loading.gif';
import './Search.css';

export class Search extends Component {
  constructor(props) {
    super(props);
    this.state = {
      games: [],
      search: '',
      game: {},
      error: '',
      loading: false,
      gameAdded: 'not-added',
    }
  }

  handleChange = e => {
    e.preventDefault();
    this.setState({
      [e.target.name]: e.target.value
    })
  }

  handleSubmit = e => {
    e.preventDefault();
    this.triggerSearch(this.state.search);
    this.setState({search: '', loading: true})
  }


  triggerSearch = async (string) => {
    const search = string.toLowerCase().split(' ').join('+');
    try {
      const result = await api.fetchBoardGames(`https://cors-anywhere.herokuapp.com/www.boardgamegeek.com/xmlapi2/search?query=${search}`)
      const games = api.cleanSearch(result);
      this.setState({ games, error: '', loading: false, gameAdded: 'not-added' });  
    } catch (error) {
      this.setState({ error: error.message, loading: false, gameAdded: 'not-added' })
    }
  }

  handleChooseGame = async (gameSelected) => {
    this.setState({loading: true})
    const { id, name } = gameSelected;
    try {
      const result = await api.fetchBoardGames(`https://cors-anywhere.herokuapp.com/https://www.boardgamegeek.com/xmlapi2/thing?id=${id}`);
      const details = await api.cleanGameDetails(result);
      const game = {...api.cleanGameDetails(result), name};
      this.setState({game, error: '', loading: false, gameAdded: 'not-added'})
    } catch (error) {
      this.setState({ error: error.message, loading: false, gameAdded: 'not-added' })
    }
  }

  addGameToFavorites = async() => {
    try {
      await db.doWriteFavoriteData(this.props.user.uid, this.state.game);
      this.setState({gameAdded: 'added'})
      const favorites = await db.getFavorites(this.props.user.uid);
      this.props.updateFavorites(favorites)
    } catch(error) {
      this.setState({ error: error.message })
    }
  }

  displayAllGames = () => {
    if(this.state.games.length)  {
      return this.state.games.map((game, index) => <h3 key={index} onClick={() => this.handleChooseGame(game)}>{game.name}</h3> )
    } else {
      return <h4 className="no-results"> No search results currently </h4>
    }
  }

  displayGame = () => {
    return this.state.game.thumbnail 
    ?  <article className="game-description">
        <button id={this.state.gameAdded} onClick={this.addGameToFavorites}></button>
        <Card favorite={this.state.game} type={"search"}/>
      </article>
    : <div></div>
  }

  displayError = () => {
    return this.state.error !== '' && <p>{this.state.error}</p>
  }

  displayLoadingGif = () => {
    return this.state.loading && <div id="loading-cont"><img src={loading} alt="loading" id="loading" /></div>
  }

  render() {
    return (
      <section className='Search'>
        { this.displayError() }
        <form onSubmit={this.handleSubmit}>
          <input 
            type="text"
            name="search"
            placeholder="Search for board games"
            value={this.state.search}
            onChange={this.handleChange}
          />
          <button type="submit">Search</button>
        </form>
        <div className="display-friends">
          <section className='search-results'>
            { this.displayLoadingGif() }
            { this.displayAllGames() }
          </section>
        </div>
        { this.displayGame() }
      </section>
    )
  }
}

Search.propTypes = {
  user: PropTypes.object.isRequired,
  updateFavorites: PropTypes.func.isRequired
};

export const mapStateToProps = state => ({
  user: state.user
})

export const mapDispatchToProps = dispatch => ({
  updateFavorites: favorites => dispatch(updateFavorites(favorites))
})

export default connect(mapStateToProps, mapDispatchToProps)(Search);