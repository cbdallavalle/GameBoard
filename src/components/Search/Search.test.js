import React from 'react';
import { Search } from './Search';
import { shallow } from 'enzyme';
import { db } from '../../firebase';
import * as api from '../../helper/bg-api-cleaner';

describe('triggerSearch', () => {
  const defaultState = {
    games: [],
    search: '',
    game: {}
  }
  const mockEvent = {
    preventDefault: () => {},
    target: {
      name: 'search',
      value: 'Meeple Circus'
    }
  }
  const mockSearch = [{id: '123', name: 'Mysterium'}]
  const mockGame = { id: '123', name: 'Mysterium', thumbnail: 'src', image: 'src', description: 'A mystery game with pictures' };
  const mockUpdateFavorites = jest.fn();

  it('should start with default state', () => {
    const wrapper = shallow(
      <Search 
        user={{}}
        updateFavorites={mockUpdateFavorites}
      />);
    expect(wrapper.state()).toEqual(defaultState);
  })

  //handleChange
  it('handleChange should set state from a mockEvent', () => {
    const wrapper = shallow(
      <Search 
        user={{}}
        updateFavorites={mockUpdateFavorites}
      />);
    wrapper.instance().handleChange(mockEvent);
    expect(wrapper.state().search).toEqual('Meeple Circus');
  })

  //handleSubmit
  it('handleSubmit should set state to an empty string', () => {
    const wrapper = shallow(
      <Search 
        user={{}}
        updateFavorites={mockUpdateFavorites}
      />);
    wrapper.instance().triggerSearch = jest.fn();
    wrapper.instance().handleSubmit(mockEvent);
    expect(wrapper.state()).toEqual(defaultState)
  })

  //triggerSearch
  it('Trigger search should set state with an array of games', async() => {
    const wrapper = shallow(
      <Search 
        user={{}}
        updateFavorites={mockUpdateFavorites}
      />);
    
    api.fetchBoardGames = jest.fn();
    api.cleanSearch = (result) => mockSearch;
    await wrapper.instance().triggerSearch('Mysterium');
    expect(wrapper.state().games).toEqual(mockSearch)
  });

  //handleChooseGame
  it('handleChooseGame should set state with a game', async () => {
    const wrapper = shallow(
      <Search 
        user={{}}
        updateFavorites={mockUpdateFavorites}
      />);

    api.fetchBoardGames = jest.fn();
    api.cleanGameDetails = (result) => (mockGame);
    await wrapper.instance().handleChooseGame(mockSearch[0])
    expect(wrapper.state().game).toEqual(mockGame)
  })

  //getGamesFromDB
  it('getGamesFromDB should return favorites', () => {
    // db.onceGetUsers() = jest.fn();
    
  })

  //addGameToFavorites

  //displayAllGames
  it('displayAllGames and displayGame should match the snapshot if games in state has length', () => {
    const wrapper = shallow(
      <Search 
        user={{}}
        updateFavorites={mockUpdateFavorites}
      />);
    wrapper.setState({games: mockSearch})
    expect(wrapper).toMatchSnapshot();
  })

  it('displayAllGames should match the snapshot if games in state has no length', () => {
    const wrapper = shallow(
      <Search 
        user={{}}
        updateFavorites={mockUpdateFavorites}
      />);
    expect(wrapper).toMatchSnapshot();
  })

  //displayGame
  it('displayGame should match the snapshot if game in state exists', () => {
    const wrapper = shallow(
      <Search 
        user={{}}
        updateFavorites={mockUpdateFavorites}
      />);
    wrapper.setState({game: mockGame});
    expect(wrapper).toMatchSnapshot();
  })
})





// describe('mapStateToProps', () => {
//   it('should map search to props', () => {
//     const mapped = mapStateToProps({ search: "Onirim"});

//     expect(mapped.search).toEqual("Onirim")
//   })
// })