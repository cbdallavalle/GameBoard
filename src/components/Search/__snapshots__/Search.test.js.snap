// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`triggerSearch displayAllGames and displayGame should match the snapshot if games in state has length 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Search
    updateFavorites={[Function]}
    user={Object {}}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <form
          onSubmit={[Function]}
>
          <input
                    name="search"
                    onChange={[Function]}
                    placeholder="Search for board games"
                    type="text"
                    value=""
          />
          <button
                    type="submit"
          >
                    Search
          </button>
</form>,
        <div
          className="display-friends"
>
          <section
                    className="search-results"
          >
                    <h3
                              onClick={[Function]}
                    >
                              Mysterium
                    </h3>
          </section>
</div>,
        <div />,
      ],
      "className": "Search",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <input
              name="search"
              onChange={[Function]}
              placeholder="Search for board games"
              type="text"
              value=""
/>,
            <button
              type="submit"
>
              Search
</button>,
          ],
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "name": "search",
              "onChange": [Function],
              "placeholder": "Search for board games",
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Search",
              "type": "submit",
            },
            "ref": null,
            "rendered": "Search",
            "type": "button",
          },
        ],
        "type": "form",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <section
            className="search-results"
>
            <h3
                        onClick={[Function]}
            >
                        Mysterium
            </h3>
</section>,
          "className": "display-friends",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h3
                onClick={[Function]}
>
                Mysterium
</h3>,
            ],
            "className": "search-results",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "host",
              "props": Object {
                "children": "Mysterium",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Mysterium",
              "type": "h3",
            },
          ],
          "type": "section",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "div",
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <form
            onSubmit={[Function]}
>
            <input
                        name="search"
                        onChange={[Function]}
                        placeholder="Search for board games"
                        type="text"
                        value=""
            />
            <button
                        type="submit"
            >
                        Search
            </button>
</form>,
          <div
            className="display-friends"
>
            <section
                        className="search-results"
            >
                        <h3
                                    onClick={[Function]}
                        >
                                    Mysterium
                        </h3>
            </section>
</div>,
          <div />,
        ],
        "className": "Search",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <input
                name="search"
                onChange={[Function]}
                placeholder="Search for board games"
                type="text"
                value=""
/>,
              <button
                type="submit"
>
                Search
</button>,
            ],
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "name": "search",
                "onChange": [Function],
                "placeholder": "Search for board games",
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Search",
                "type": "submit",
              },
              "ref": null,
              "rendered": "Search",
              "type": "button",
            },
          ],
          "type": "form",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <section
              className="search-results"
>
              <h3
                            onClick={[Function]}
              >
                            Mysterium
              </h3>
</section>,
            "className": "display-friends",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h3
                  onClick={[Function]}
>
                  Mysterium
</h3>,
              ],
              "className": "search-results",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": "0",
                "nodeType": "host",
                "props": Object {
                  "children": "Mysterium",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "Mysterium",
                "type": "h3",
              },
            ],
            "type": "section",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "div",
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`triggerSearch displayAllGames should match the snapshot if games in state has no length 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Search
    updateFavorites={[Function]}
    user={Object {}}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <form
          onSubmit={[Function]}
>
          <input
                    name="search"
                    onChange={[Function]}
                    placeholder="Search for board games"
                    type="text"
                    value=""
          />
          <button
                    type="submit"
          >
                    Search
          </button>
</form>,
        <div
          className="display-friends"
>
          <section
                    className="search-results"
          >
                    <h4>
                               No search results currently 
                    </h4>
          </section>
</div>,
        <div />,
      ],
      "className": "Search",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <input
              name="search"
              onChange={[Function]}
              placeholder="Search for board games"
              type="text"
              value=""
/>,
            <button
              type="submit"
>
              Search
</button>,
          ],
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "name": "search",
              "onChange": [Function],
              "placeholder": "Search for board games",
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Search",
              "type": "submit",
            },
            "ref": null,
            "rendered": "Search",
            "type": "button",
          },
        ],
        "type": "form",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <section
            className="search-results"
>
            <h4>
                         No search results currently 
            </h4>
</section>,
          "className": "display-friends",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <h4>
               No search results currently 
</h4>,
            "className": "search-results",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": " No search results currently ",
            },
            "ref": null,
            "rendered": " No search results currently ",
            "type": "h4",
          },
          "type": "section",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "div",
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <form
            onSubmit={[Function]}
>
            <input
                        name="search"
                        onChange={[Function]}
                        placeholder="Search for board games"
                        type="text"
                        value=""
            />
            <button
                        type="submit"
            >
                        Search
            </button>
</form>,
          <div
            className="display-friends"
>
            <section
                        className="search-results"
            >
                        <h4>
                                     No search results currently 
                        </h4>
            </section>
</div>,
          <div />,
        ],
        "className": "Search",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <input
                name="search"
                onChange={[Function]}
                placeholder="Search for board games"
                type="text"
                value=""
/>,
              <button
                type="submit"
>
                Search
</button>,
            ],
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "name": "search",
                "onChange": [Function],
                "placeholder": "Search for board games",
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Search",
                "type": "submit",
              },
              "ref": null,
              "rendered": "Search",
              "type": "button",
            },
          ],
          "type": "form",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <section
              className="search-results"
>
              <h4>
                             No search results currently 
              </h4>
</section>,
            "className": "display-friends",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <h4>
                 No search results currently 
</h4>,
              "className": "search-results",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": " No search results currently ",
              },
              "ref": null,
              "rendered": " No search results currently ",
              "type": "h4",
            },
            "type": "section",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "div",
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`triggerSearch displayGame should match the snapshot if game in state exists 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Search
    updateFavorites={[Function]}
    user={Object {}}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <form
          onSubmit={[Function]}
>
          <input
                    name="search"
                    onChange={[Function]}
                    placeholder="Search for board games"
                    type="text"
                    value=""
          />
          <button
                    type="submit"
          >
                    Search
          </button>
</form>,
        <div
          className="display-friends"
>
          <section
                    className="search-results"
          >
                    <h4>
                               No search results currently 
                    </h4>
          </section>
</div>,
        <article
          className="game-description"
>
          <button
                    id="add-game"
                    onClick={[Function]}
          >
                    <i
                              className="fas fa-plus"
                    />
          </button>
          <Connect(Card)
                    display="hide"
                    favorite={
                              Object {
                                        "description": "A mystery game with pictures",
                                        "id": "123",
                                        "image": "src",
                                        "name": "Mysterium",
                                        "thumbnail": "src",
                                      }
                    }
          />
</article>,
      ],
      "className": "Search",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <input
              name="search"
              onChange={[Function]}
              placeholder="Search for board games"
              type="text"
              value=""
/>,
            <button
              type="submit"
>
              Search
</button>,
          ],
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "name": "search",
              "onChange": [Function],
              "placeholder": "Search for board games",
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Search",
              "type": "submit",
            },
            "ref": null,
            "rendered": "Search",
            "type": "button",
          },
        ],
        "type": "form",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <section
            className="search-results"
>
            <h4>
                         No search results currently 
            </h4>
</section>,
          "className": "display-friends",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <h4>
               No search results currently 
</h4>,
            "className": "search-results",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": " No search results currently ",
            },
            "ref": null,
            "rendered": " No search results currently ",
            "type": "h4",
          },
          "type": "section",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <button
              id="add-game"
              onClick={[Function]}
>
              <i
                            className="fas fa-plus"
              />
</button>,
            <Connect(Card)
              display="hide"
              favorite={
                            Object {
                                          "description": "A mystery game with pictures",
                                          "id": "123",
                                          "image": "src",
                                          "name": "Mysterium",
                                          "thumbnail": "src",
                                        }
              }
/>,
          ],
          "className": "game-description",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <i
                className="fas fa-plus"
/>,
              "id": "add-game",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "fas fa-plus",
              },
              "ref": null,
              "rendered": null,
              "type": "i",
            },
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "display": "hide",
              "favorite": Object {
                "description": "A mystery game with pictures",
                "id": "123",
                "image": "src",
                "name": "Mysterium",
                "thumbnail": "src",
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "article",
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <form
            onSubmit={[Function]}
>
            <input
                        name="search"
                        onChange={[Function]}
                        placeholder="Search for board games"
                        type="text"
                        value=""
            />
            <button
                        type="submit"
            >
                        Search
            </button>
</form>,
          <div
            className="display-friends"
>
            <section
                        className="search-results"
            >
                        <h4>
                                     No search results currently 
                        </h4>
            </section>
</div>,
          <article
            className="game-description"
>
            <button
                        id="add-game"
                        onClick={[Function]}
            >
                        <i
                                    className="fas fa-plus"
                        />
            </button>
            <Connect(Card)
                        display="hide"
                        favorite={
                                    Object {
                                                "description": "A mystery game with pictures",
                                                "id": "123",
                                                "image": "src",
                                                "name": "Mysterium",
                                                "thumbnail": "src",
                                              }
                        }
            />
</article>,
        ],
        "className": "Search",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <input
                name="search"
                onChange={[Function]}
                placeholder="Search for board games"
                type="text"
                value=""
/>,
              <button
                type="submit"
>
                Search
</button>,
            ],
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "name": "search",
                "onChange": [Function],
                "placeholder": "Search for board games",
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Search",
                "type": "submit",
              },
              "ref": null,
              "rendered": "Search",
              "type": "button",
            },
          ],
          "type": "form",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <section
              className="search-results"
>
              <h4>
                             No search results currently 
              </h4>
</section>,
            "className": "display-friends",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <h4>
                 No search results currently 
</h4>,
              "className": "search-results",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": " No search results currently ",
              },
              "ref": null,
              "rendered": " No search results currently ",
              "type": "h4",
            },
            "type": "section",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <button
                id="add-game"
                onClick={[Function]}
>
                <i
                                className="fas fa-plus"
                />
</button>,
              <Connect(Card)
                display="hide"
                favorite={
                                Object {
                                                "description": "A mystery game with pictures",
                                                "id": "123",
                                                "image": "src",
                                                "name": "Mysterium",
                                                "thumbnail": "src",
                                              }
                }
/>,
            ],
            "className": "game-description",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <i
                  className="fas fa-plus"
/>,
                "id": "add-game",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "fas fa-plus",
                },
                "ref": null,
                "rendered": null,
                "type": "i",
              },
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "display": "hide",
                "favorite": Object {
                  "description": "A mystery game with pictures",
                  "id": "123",
                  "image": "src",
                  "name": "Mysterium",
                  "thumbnail": "src",
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "article",
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
